homeassistant/components/zha/__init__.py

```diff
@@ -435,3 +435,42 @@ def get_discovery_info(hass, discovery_info):
         return result
     except Exception:  # pylint: disable=broad-except
         return {}
+
+
+async def configure_reporting(entity_id, cluster, attr, skip_bind=False,
+                              min_report=300, max_report=900,
+                              reportable_change=1):
+    """Configure attribute reporting for a cluster.
+
+    while swallowing the DeliverError exceptions in case of unreachable
+    devices.
+    """
+    from zigpy.exceptions import DeliveryError
+
+    attr_name = cluster.attributes.get(attr, [attr])[0]
+    cluster_name = cluster.ep_attribute
+    if not skip_bind:
+        try:
+            res = await cluster.bind()
+            _LOGGER.debug(
+                "%s: bound  '%s' cluster: %s", entity_id, cluster_name, res[0]
+            )
+        except DeliveryError as ex:
```

Comment:
Should the method stop execution if this fails?

Reply:
It is the same idea as behind [`zha.safe_read()`](https://github.com/home-assistant/home-assistant/blob/2682d26f79c383fd623d82adb2a26afa041d7605/homeassistant/components/zha/__init__.py#L390) Better to have an entity which may not push external updates, but still works otherwise and could be controlled, rather than no entity at all.
Some vendors (eg Xiaomi) will throw, but still would send attribute reports.

Reply:
In the long run I'm thinking about having it as a "service call" on a device, like in a "repair_device" service call